bliss/encoder.py

```diff
@@ -123,15 +123,25 @@ def encode_batch(self, batch):
 
         return pred
 
-    def variational_mode(self, pred: Dict[str, Tensor]) -> Dict[str, Tensor]:
-        """Compute the mode of the variational distribution."""
+    def variational_mode(
+        self, pred: Dict[str, Tensor], return_full: Optional[bool] = True
+    ) -> Union[FullCatalog, TileCatalog]:
+        """Compute the mode of the variational distribution.
+
+        Args:
+            pred (Dict[str, Tensor]): model predictions
+            return_full (bool, optional): If True, returns a FullCatalog instead of a TileCatalog.
+                Defaults to False.
+
+        Returns:
+            Union[FullCatalog, TileCatalog]: Catalog based on predictions.
+        """
         # the mean would be better at minimizing squared error...should we return that instead?
         tile_is_on_array = pred["on_prob"].mode
         # this is the mode of star_log_flux but not the mean of the star_flux distribution
         star_fluxes = pred["star_log_flux"].mode.exp()  # type: ignore
-        star_fluxes *= tile_is_on_array
-        galaxy_bools = pred["galaxy_prob"].mode * pred["on_prob"].mode  # type: ignore
-        star_bools = (1 - pred["galaxy_prob"].mode) * pred["on_prob"].mode  # type: ignore
```

Comment:
Could this change break other parts of the code? If you don't multiple by `pred["on_prob"].mode` here, do we use `est_cat.n_sources` everywhere else to determine what stars/galaxies have been detected?

Reply:
`TileCatalog.to_full_params` also masks by `is_on_array`, so doing it here was unnecessary (and also not very useful, since none of the galaxy params were masked anyway). I confirmed that the FullCatalog returned by calling variational_mode is the same before and after the changes. 
https://github.com/prob-ml/bliss/blob/bc0e2df666c9672107e5a03861953ca0c6cf5bc3/bliss/catalog.py#L117-L119

Detection metrics use `n_sources` to slice `true_locs` and `est_locs` (since they're sorted with tiles being on at the top):
https://github.com/prob-ml/bliss/blob/bc0e2df666c9672107e5a03861953ca0c6cf5bc3/bliss/metrics.py#L150-L152

Classification metrics also mask by the tile being on and the source being a galaxy/star in the true catalog:
https://github.com/prob-ml/bliss/blob/bc0e2df666c9672107e5a03861953ca0c6cf5bc3/bliss/metrics.py#L195-L202

I think these are the only places that use the output of variational mode.